src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -81,21 +87,12 @@ public BackgroundCertificateRefresh(
      * Start running the task every DEFAULT_INTERVAL_SECONDS.
      */
     public void start() {
-        start(DEFAULT_INTERVAL_SECONDS);
-    }
-
-    /**
-     * Start running the task on every intervalSeconds.
-     * @param intervalSeconds - frequency for this task to run
-     */
-    public void start(int intervalSeconds) {
         if (scheduledFuture != null) {
-           return;
+            return;
         }
 
-        logger.info("Starting background refresh of client certificates every {} seconds", intervalSeconds);
-        scheduledFuture =
-            scheduler.scheduleAtFixedRate(this, intervalSeconds, intervalSeconds, TimeUnit.SECONDS);
+        logger.info("Starting background refresh of client certificates every {} seconds", DEFAULT_INTERVAL_SECONDS);
+        scheduleNextRun();
```

Comment:
Did you consider running immediately on boot instead of scheduling the first refresh in 24 hours? I think we could defend either. Curious about your thoughts.

Reply:
The main reason I did not do it on bootup is because a lot of being able to refresh certificates relies on the fact that we have the certificate PEMs stored. Given the current implementation stores things in memory (for the pems) if we can't find the PEMs then we can't update them